src/Microsoft.AspNetCore.SystemWebAdapters.CoreServices/HttpApplication/HttpApplicationPooledObjectPolicy.cs

```diff
@@ -0,0 +1,234 @@
+        { "Application_ReleaseRequestState", (app, handler) => app.ReleaseRequestState += handler },
+
+        // Fired when the ASP.NET page framework completes an authorization request. It allows caching modules to serve the request from the cache, thus bypassing handler execution.
+        { "Application_ResolveRequestCache", (app, handler) => app.ResolveRequestCache += handler },
+
+        // Fired when the ASP.NET page framework completes handler execution to allow caching modules to store responses to be used to handle subsequent requests.
+        { "Application_UpdateRequestCache", (app, handler) => app.UpdateRequestCache += handler },
+
+        // Fired when the security module has established the current user's identity as valid. At this point, the user's credentials have been validated.
+        { "Application_AuthenticateRequest", (app, handler) => app.AuthenticateRequest += handler },
+
+        // Fired when the security module has verified that a user can access resources.
+        { "Application_AuthorizeRequest", (app, handler) => app.AuthorizeRequest += handler },
+
+        // Fired when a new user visits the application Web site.
+        { "Session_Start", (app, handler) => app.SessionStart += handler },
+
+        // Fired when a user's session times out, ends, or they leave the application Web site.
+        { "Session_End", (app, handler) => app.SessionEnd += handler },
+    };
+
+    [LoggerMessage(0, LogLevel.Information, "Registered event {ApplicationType}.{EventName}")]
+    private partial void LogRegistration(string applicationType, string eventName);
+
+    [LoggerMessage(1, LogLevel.Warning, "HttpApplication event {ApplicationType}.{EventName} is unsupported")]
+    private partial void LogUnsupported(string applicationType, string eventName);
+
+    [LoggerMessage(2, LogLevel.Warning, "{ApplicationType}.{EventName} has unsupported signature")]
+    private partial void LogInvalid(string applicationType, string eventName);
+
+    private readonly ILogger<HttpApplicationPooledObjectPolicy> _logger;
+    private readonly IServiceProvider _services;
+    private readonly Lazy<Func<IServiceProvider, HttpApplication>> _factory;
+    private readonly HttpApplicationState _state;
+
+    public HttpApplicationPooledObjectPolicy(IServiceProvider services, IOptions<HttpApplicationOptions> options, ILogger<HttpApplicationPooledObjectPolicy> logger)
+    {
+        _logger = logger;
+        _services = services;
+        _state = new HttpApplicationState();
+        _factory = new Lazy<Func<IServiceProvider, HttpApplication>>(() => CreateFactory(options.Value), isThreadSafe: true);
+    }
+
+    public override HttpApplication Create()
+    {
+        var app = _factory.Value(_services);
+
+        // This is invoked each time an HttpApplication is constructed
+        app.InvokeEvent(ApplicationEvent.ApplicationInit);
```

Comment:
I didn't realize this was an event. It would be useful to comment to the reference source or documentation about these things.

Reply:
Makes sense. I had a link at the feature level, but not here at the implementation level. Added a comment on the class so it's in the file and some links to equivalent stuff in reference source